Automattic/jetpack
# Boost: Fix ISA showing an error if no report was found

Fixes an error in the ISA description on freshly installed premium sites:

![image](https://github.com/user-attachments/assets/3d0bac8f-ac44-460f-8e87-27d98dd77ac7)

The reason this happens is because the UI makes a request for the `latest` report to WP.com. There's no report, and WP.com reports `not-found` for that report. Due to [a fix of a bug](https://github.com/Automattic/jetpack/pull/40096/files#diff-679b5f5ab1f61e24536ce8efbd2d1bccc7bfc91f4bba48ef82b158704a7c89ceL19) in a previous PR, this started showing as an error since Boost 3.6.0 was released.

This changes the UI to ignore the `not-found` status from WP.com (just like it did prior to the fixing the above mentioned bug).

## Proposed changes:

* Update Boost API client to support responses from WordPress.com ISA endpoints;
* Fix `not-found` reports showing as an error in the UI.

### Other information:


## Jetpack product discussion

p1734518619728809-slack-C016BBAFHHS

## Does this pull request change what data or activity we track or use?

no

## Testing instructions:

### Test that you can see an error:

- Setup Boost premium from trunk (or latest production version - 3.6.1);
- Open the Boost dashboard - there should be an error under "Image Size Analysis" like the one from the screenshot above.

### Test that the error is not showing:
- Setup Boost premium (this branch);
- Open the Boost dashboard. There should be no error under "Image Size Analysis".
